projects/observability/src/shared/graphql/model/specifications/composite-specification.ts

```diff
@@ -2,5 +2,5 @@ import { Dictionary } from '@hypertrace/common';
 import { Specification } from '../schema/specifier/specification';
 
 export interface CompositeSpecification extends Specification {
-  extractFromServerData(resultContainer: Dictionary<unknown>): unknown[];
+  extractFromServerData(resultContainer: Dictionary<unknown>): unknown[] | Dictionary<unknown>;
```

Comment:
we should consider using generic typing here.

Reply:
Yeah, tried that also, didn't work that well